x/dex/keeper/cancel_limit_order.go

```diff
@@ -88,8 +88,8 @@ func (k Keeper) ExecuteCancelLimitOrder(
 		return sdk.Coin{}, sdk.Coin{}, nil, sdkerrors.Wrapf(types.ErrCancelEmptyLimitOrder, "%s", tranche.Key.TrancheKey)
 	}
 
-	k.SaveTrancheUser(ctx, trancheUser)
-	k.SaveTranche(ctx, tranche)
+	k.SaveOrRemoveTrancheUser(ctx, trancheUser)
```

Comment:
I mean, this is much more confusing for me now

Can't it be two separate methods?

Reply:
Like, i understand there is a pretty simple condition inside the method, but i think the code would be much simpler if the condition is outside of the method

What do you think?